### PR TITLE
Backport VPC and GKE to terraform

### PR DESCRIPTION
**Pull request by eduartua**
* Cloudkite terraform modules used (vpc, GKE)
This solves issues:
#8 
#9

### CODE REVIEW

**Comment by victortrac** [comment]
@idvoretskyi This is terraform for making a dev environment. It's already been applied (meaning, there's a dev VPC & GKE cluster running). The plan is to use this same terraform to create a new parallel prod environment, deploy the discovery service, and then update DNS once it's been fully tested. Then we'll tear down the old prod GKE cluster.

**Comment by philips** [comment]
@victortrac How are you going to migrate the data? There are two issues with migrating to a new cluster:

1. DNS split horizon for the length of the domain TTL will cause boot strapping to fail
2. All of the data will be lost so any cluster mid-bootstrap will fail

If doing this migration is the only option please do send an email to etcd-dev@googlegroups.com to give everyone a heads up.

I can't review the terraform code however, I don't know it well enough anymore.

**Comment by victortrac** [comment]
@philips You're right about these two issues. 

1. There's a 10 min TTL on the A record for discovery.etcd.io at the moment. We can reduce to to 60s prior to the migration to reduce the likelihood of this happening, but it's not going to completely eliminate the issue. Also, my intuition says that cluster members will be using the same DNS server (since they'll likely be in the same datacenter), so it seems exceedingly unlikely that a member will resolve an IP address prior to TTL expiration (with the old IP) and then another member will grab the new IP after expiration.
2. I can see this happening for a very slow to bootstrap cluster. If a member starts and resolves the old discovery service and then subsequent members don't bootstrap until after the TTL refreshes to the new discovery service, then this will happen. Again, this seems exceedingly unlikely if end-users are using automated processes to bootstrap the cluster, since the cluster members will likely come up fairly simultaneously.

The traffic is fairly low, and we'll do the DNS swap during low-traffic periods. We'll also notify the dev mailing list. There are (time) expensive technical ways to possibly mitigate these concerns further, but they don't seem worth it to me. Does this address your concerns?

**Comment by philips** [comment]
Yes, sounds good.

File: terraform/dev/main.tf [hunk 1]
```diff
@@ -24,9 +24,11 @@ module "gke" {
   gke_nodepools = var.gke_nodepools
```
Comment (victortrac) [inline terraform/dev/main.tf]: What is line 22 and 23 doing?

File: terraform/dev/main.tf [hunk 2]
```diff
@@ -24,9 +24,11 @@ module "gke" {
...
   gke_pods_secondary_range_name = module.vpc.gke_subnetwork_secondary_range_name_services
   gke_services_secondary_range_name = module.vpc.gke_subnetwork_secondary_range_name_pods
-  network = module.vpc.network
   subnetwork = module.vpc.gke_subnetwork
   gke_master_authorized_networks = var.gke_master_authorized_networks
+  master_ipv4_cidr_block = var.gke["master_ipv4_cidr_block"]
+  project = var.gcp["project"]
+  region = var.gcp["region"]
```
Comment (victortrac) [inline terraform/dev/main.tf]: Do we still want gke and gcp to be maps?  What are their expected keys?

**Pull request merged**